src/kernels/maha.jl

```diff
@@ -0,0 +1,22 @@
+"""
+    MahaKernel(P::AbstractMatrix, kappa::Function)
+
+    Mahalanobis distance-based kernel given by
+```math
+    κ(x,y) =  kappa(r^2), r^2 = maha(x,P,y) = (x-y)'*inv(P)*(x-y)
+```
+where the matrix P is the metric and kappa is a user defined kernel function.
+
+"""
+struct MahaKernel{T<:Real} <: BaseKernel
+    P::AbstractMatrix{T}
+    kappa::Function
```

Comment:
This will lead to bad performance, see https://docs.julialang.org/en/v1/manual/performance-tips/index.html#Avoid-fields-with-abstract-type-1. Please change it to
```suggestion
struct MahaKernel{T<:Real,A<:AbstractMatrix{T},F} <: BaseKernel
    P::A
    kappa::F
```

I'm also wondering if we actually want to allow a flexible `kappa`. IIRC this would be the first kernel where you could specify a positive definite function, so it does not seem completely in line with the current design. What's the general opinion on this?